node/src/components/consensus/highway_core/highway_testing.rs

```diff
@@ -541,7 +541,15 @@ where
                     {
                         Err((pvv, error)) => return Ok(Err((pvv.into_vertex(), error))),
                         Ok(valid_vertex) => self.call_validator(rng, &recipient, |v, rng| {
-                            v.highway_mut().add_valid_vertex(valid_vertex, rng)
+                            // TODO: Find a better way of making deterministic timestamps for
+                            // testing than this kluge
+                            let now = match &valid_vertex {
+                                ValidVertex(Vertex::Vote(signed_wire_vote)) => {
+                                    signed_wire_vote.wire_vote.timestamp
+                                }
+                                ValidVertex(Vertex::Evidence(_)) => 0.into(),
+                            };
```

Comment:
@afck This should just be the heuristic logic that was there before we put in `Timestamp::now()`

Reply:
Actually, at the call site of `process_message`, we still know the (simulated) `delivery_time` and could pass it in. Then pass it into `add_vertex` and we should have the "correct" timestamp here.

Reply:
Done.